src/software/ai/evaluation/calc_best_shot.cpp

```diff
@@ -24,36 +97,51 @@ std::optional<Shot> calcBestShotOnGoal(const Field &field, const Team &friendly_
                                        double radius)
 {
     std::vector<Robot> obstacles;
-    for (const Robot &enemy_robot : enemy_team.getAllRobots())
-    {
-        // Only add the robot to the obstacles if it is not ignored
-        if (std::count(robots_to_ignore.begin(), robots_to_ignore.end(), enemy_robot) ==
-            0)
-        {
-            obstacles.emplace_back(enemy_robot);
-        }
-    }
-    for (const Robot &friendly_robot : friendly_team.getAllRobots())
+    std::vector<Robot> all_robots;
+
+    size_t max_num_robots =
+        (enemy_team.numRobots() + friendly_team.numRobots()) <= 1
+            ? 1
+            : (enemy_team.numRobots() + friendly_team.numRobots() - 1);
+    all_robots.reserve(max_num_robots);
+    all_robots.insert(all_robots.begin(), enemy_team.getAllRobots().begin(),
+                      enemy_team.getAllRobots().end());
+    all_robots.insert(all_robots.begin(), friendly_team.getAllRobots().begin(),
+                      friendly_team.getAllRobots().end());
+
+    for (const Robot &robot : all_robots)
     {
-        // Only add the robot to the obstacles if it is not ignored
-        if (std::count(robots_to_ignore.begin(), robots_to_ignore.end(),
-                       friendly_robot) == 0)
+        if (std::count(robots_to_ignore.begin(), robots_to_ignore.end(), robot) == 0)
         {
-            obstacles.emplace_back(friendly_robot);
+            if (goal == TeamType::ENEMY)
```

Comment:
very clever! can get rid of a bunch of obstacles with this

Reply:
:D ya I don't think there is a case where it will lose resolution because of this